src/ngraph/pass/algebraic_simplification.cpp

```diff
@@ -345,6 +347,86 @@ static bool is_input_uniform_constant(shared_ptr<Node> op,
     return rc;
 }
 
+//`simplify_gather`, optimizes gather if Gather is gathering the
+// whole input tensor
+static bool simplify_gather(std::shared_ptr<Node> node)
+{
+    auto gather = as_type_ptr<op::v1::Gather>(node);
+
+    if (gather)
+    {
+        // check if we are gathering the whole input
+        auto data = node->get_argument(0);
```

Comment:
These should be `input_value(0)`, `input_value(1)` since `get_argument` doesn't work for multi-output nodes.